aten/src/ATen/native/ForeachOpsKernels.cpp

```diff
@@ -107,29 +107,55 @@ void foreach_tensor_##NAME##_slow_(TensorList tensors) {                   \
+                                                                                                                                     \
+void foreach_tensor_##NAME##_scalar_slow_(TensorList input, TensorList tensors1, TensorList tensors2, Scalar scalar) {               \
+  TORCH_CHECK(input.size() > 0, "Tensor list must have at least one tensor.");                                                       \
+  TORCH_CHECK(input.size() == tensors1.size(), "Tensor lists must be of the same length.");                                          \
+  TORCH_CHECK(tensors1.size() == tensors2.size(), "Tensor lists must be of the same length.");                                       \
+                                                                                                                                     \
+  for (int i = 0; i < input.size(); i++) {                                                                                           \
+    input[i].NAME##_(tensors1[i], tensors2[i], scalar);                                                                              \
+  }                                                                                                                                  \
+}                                                                                                                                    \
+
+#define FOREACH_POINTWISE_OP_SCALARLIST(NAME)                                                                                                           \
+std::vector<Tensor> foreach_tensor_##NAME##_scalarlist_slow(TensorList input, TensorList tensors1, TensorList tensors2, at::ArrayRef<double> scalars) { \
+  TORCH_CHECK(input.size() > 0, "Tensor list must have at least one tensor.");                                                                          \
+  TORCH_CHECK(input.size() == tensors1.size(), "Tensor lists must be of the same length.");                                                             \
+  TORCH_CHECK(tensors1.size() == tensors2.size(), "Tensor lists must be of the same length.");                                                          \
+  check_foreach_api_restrictions(tensors1, scalars);                                                                                                    \
+                                                                                                                                                        \
+  std::vector<Tensor> result;                                                                                                                           \
+  for (int i = 0; i < input.size(); i++) {                                                                                                              \
+    result.emplace_back(input[i].NAME(tensors1[i], tensors2[i], scalars[i]));                                                                           \
```

Comment:
nit: OP or FUNC seems more descriptive than NAME, but this seems like a pre-existing problem